tests/agent_features/test_ignore_expected_errors.py

```diff
@@ -156,6 +157,61 @@ def _test():
     _test()
 
 
+# =============== Test error trace attributes ===============
+
+
+error_trace_settings_matrix = [
+    ({}, False),
+    (expected_runtime_error_settings, True),
+]
+override_expected_matrix = (True, False, None)
+
+
+@pytest.mark.parametrize("settings,expected", error_trace_settings_matrix)
+@pytest.mark.parametrize("override_expected", override_expected_matrix)
+def test_error_trace_attributes_inside_transaction(
+    settings, expected, override_expected
+):
+    expected = override_expected if override_expected is not None else expected
+
+    # Set attributes from parameters
+    error_trace_attributes = {
+        "intrinsic": {"error.expected": expected},
+        "agent": {},
+        "user": {},
+    }
+
+    @validate_transaction_error_trace_attributes(error_trace_attributes)
+    @background_task(name="test")
+    @override_application_settings(settings)
+    def _test():
+        exercise(override_expected)
+
+    _test()
+
+
+@pytest.mark.parametrize("settings,expected", error_trace_settings_matrix)
+@pytest.mark.parametrize("override_expected", override_expected_matrix)
+def test_error_trace_attributes_outside_transaction(
+    settings, expected, override_expected
+):
+    expected = override_expected if override_expected is not None else expected
+
+    # Set attributes from parameters
+    error_trace_attributes = {
+        "intrinsic": {"error.expected": expected},
+        "agent": {},
+        "user": {},
+    }
+
```

Comment:
We have a validator called `validate_error_trace_attributes_outside_transaction` - I'm wondering if we would want to use that here.

Reply:
Done